admin-dev/themes/new-theme/js/components/grid/extension/ajax-bulk-action-extension.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import {Grid} from '@PSTypes/grid';
+import ProgressModal from '@components/modal/progress-modal';
+import GridMap from '@components/grid/grid-map';
+
+const {$} = window;
+
+/**
+ * Handles submit of grid actions
+ */
+export default class AjaxBulkActionExtension {
+  /**
+   * Extend grid with bulk action submitting
+   *
+   * @param {Grid} grid
+   */
+  extend(grid: Grid): void {
+    grid
+      .getContainer()
+      .on('click', GridMap.bulks.ajaxAction, (event: JQueryEventObject) => {
+
+        let total = $('.js-bulk-action-checkbox:checked').length;
+
+        const $ajaxButton = $(event.currentTarget);
+
+        const modal = this.showProgressModal(
+          $ajaxButton,
+          grid,
+          total,
+        );
+        let Id;
+        let doneCount = 0;
+        $('.js-bulk-action-checkbox:checked').each( function (i) {
```

Comment:
Hi @JevgenijVisockij 

here's an interesting alternative I discovered in @zuk3975 PR to handle the request loop:

https://github.com/zuk3975/PrestaShop/blob/d2a791d501a9766e3c4b85f699a194eba11ff64c/admin-dev/themes/new-theme/js/pages/product/combination/bulk-form-handler.ts#L166-L180

Instead of using a recursive callback in the request success/error callback you can use `await` on an `async` function responsible for the request, thus making the whole loop synchronous instead!